index.js

```diff
@@ -56,24 +64,50 @@ async function start () {
 
   tracking.swarm = swarm
 
+  swarm.on('connection', onsocket)
+  swarm.listen()
+
+  const lists = [].concat(argv.list || [])
   const cores = [].concat(argv.key || [])
   const bees = [].concat(argv.bee || [])
   const drives = [].concat(argv.drive || [])
   const seeders = [].concat(argv.seeder || [])
 
   if (argv.file) {
-    const seeds = await fs.promises.readFile(argv.file)
-    for (const [type, key] of configs.parse(seeds, { split: ' ', length: 2 })) {
-      if (type === 'key') cores.push(key)
+    const file = await fs.promises.readFile(argv.file)
+    const list = configs.parse(file, { split: ' ', length: 2 })
+
+    for (const [type, key] of list) {
+      // TODO: simplify
+      if (type === 'list') lists.push(key)
+      else if (type === 'key') cores.push(key)
       else if (type === 'bee') bees.push(key)
       else if (type === 'bundle' || type === 'drive') drives.push(key)
       else if (type === 'seeder') seeders.push(key)
-      else throw new Error('Invalid seed type: ' + type)
+      else throw new Error('Invalid seed type: ' + type + ' for ' + key)
     }
   }
 
-  swarm.on('connection', onsocket)
-  swarm.listen()
+  for (const key of lists) {
+    const bee = await downloadLists(key)
+
+    if (argv['dry-run']) continue
+
+    for (const type of ['core', 'bee', 'drive', 'seeder']) {
```

Comment:
instead of using a sub for each, can we have the type in the value instead? then you just need to know the key for interact with it

Reply:
and then have no subs